### DISCUSSION TITLE
Why is there no "as" for HasMany/BelongsToMany in v7?

### DISCUSSION THREAD

**Discussion by pocketarc** [sequelize/sequelize#17212] [Help & Questions] [answered]
In v7, let's say I have the following example:

```ts
@HasMany(() => PostComments, {
    foreignKey: "post_id",
    as: "comments",
})
```

I get the following:

```
TS2353: Object literal may only specify known properties, and as does not exist in type
Omit<HasManyOptions<string, keyof PostCommentAttributes>, "as">
```

This happens with BelongsToMany as well. 

Without the `as`, the key will be `postComments`, which is a bit more verbose than I'd like.

Why is the `as` omitted for these relationships? Is there no way to specify a custom key for these relationships?

**Comment by ephys** [comment, marked as answer]
Because the name of the association is also the name of the class field that will be used to access the eagerly loaded association, when using decorators, the name of the class field being decorated is automatically used as the `as` option

**Comment by pocketarc** [comment]
Oh, thank you! That's amazing, so if I have:

```ts
@HasMany(() => PostComments, "post_id")
declare comments?: NonAttribute<PostComments[]>;
```

It automatically knows that the "as" is "comments"! That's SO nice.

**Reply by ephys** [reply]
Exactly! :)
